### PR TITLE
feat(setServiceUrl): use setServiceUrl instead of setServiceEndpoint

### PR DESCRIPTION
**Pull request by mamoonraja**
BREAKING CHANGE: Use SetServiceUrl to set the service endpoint rather than SetEndpoint

### CODE REVIEW

**Review by mediumTaj — APPROVED** [review]
👍 looks good! tagged phil on review as well

**Review by padamstx — CHANGES_REQUESTED** [review]
Before the service_url value is used, we'll need to validate it and possibly throw/raise an error if it's not valid.  And we should add tests that check this as well :)

**Comment by mediumTaj** [comment]
For Unity we didn't have a testing platform until very recently. We were only able to do integration tests. We're moving to the new way of testing as part of the unit test generation epic. Even now though we cannot test Unity in CI since we are being blocked from downloading/installing the Unity application in travis. Also no way of finding code coverage...yet :)

File: Connection/RESTConnector.cs
Comment (padamstx) [inline Connection/RESTConnector.cs]: We also need to make sure it's not null/nil or "".   In other words, we need to do what we can to ensure that by the time we want to use the serviceUrl value to connect to the server, it is as valid as we can determine :)

**Review by padamstx — APPROVED** [review]
I left a comment which might or might not lead to additional changes, but I'll let you decide :)

File: Connection/RESTConnector.cs
Comment (mamoonraja) [reply Connection/RESTConnector.cs]: Fair enough, I didn't add it because we are setting it to default URL if it's null or empty. But probably better to keep the check anyways. Let me update.

Comment (padamstx) [reply Connection/RESTConnector.cs]: yeah, the reason why this is important to validate against null/"" here is that after constructing the service, the app could set the URL to null or "".
In the parameterized url scenario (aka full iam) there won't be a default base url at all.

File: Authentication/Iam/IamAuthenticator.cs
Comment (padamstx) [inline Authentication/Iam/IamAuthenticator.cs]: I don't want to hold up this PR, but in the other cores, this is a validation error that would prevent the authenticator from being initialized :)

File: Connection/RESTConnector.cs
Comment (padamstx) [inline Connection/RESTConnector.cs]: Minor nit: `Url` should probably be `service URL` or `serviceUrl` :)

File: Connection/RESTConnector.cs
Comment (padamstx) [inline Connection/RESTConnector.cs]: s/Url/serviceUrl/

**Review by padamstx — APPROVED** [review]
A couple things that you can choose to address or not.   Looks good.

File: Connection/RESTConnector.cs
Comment (mamoonraja) [reply Connection/RESTConnector.cs]: Since this connector will also be used by IAM Authenticator that's why I kept the name generic.

File: Authentication/Iam/IamAuthenticator.cs
Comment (mamoonraja) [reply Authentication/Iam/IamAuthenticator.cs]: it should be throwing an exception.

**Comment by padamstx** [comment]
Just realized that we need a `GetServiceUrl()` method on the BaseService class :)

**Review by padamstx — CHANGES_REQUESTED** [review]
Sorry, but just realized we need a GetServiceUrl() method on BaseService :)

**Review by padamstx — APPROVED** [review]
Looks good now.

**Pull request merged**
